src/include/duckdb.h

```diff
@@ -4485,6 +4528,69 @@ DUCKDB_C_API char *duckdb_table_description_get_column_name(duckdb_table_descrip
 // Arrow Interface
 //===--------------------------------------------------------------------===//
 
+/*!
+Transforms a DuckDB Schema into an Arrow Schema
+
+* @param client_properties The client properties to extract the arrow settings from.
+* @param types The DuckDB Logical Types for each column in the schema.
+* @param names The names for each column in the schema.
+* @param column_count The number of columns that exist in the schema.
+* @param out_schema The resulting arrow schema. Must be destroyed with `out_schema->release(out_schema)`.
+* @return The error data.
+*/
+DUCKDB_C_API duckdb_error_data duckdb_to_arrow_schema(duckdb_client_properties client_properties,
+                                                      duckdb_logical_type *types, char **names, idx_t column_count,
+                                                      struct ArrowSchema *out_schema);
+
+/*!
+Transforms a DuckDB data chunk into an Arrow array.
+
+* @param client_properties The client properties to extract the Arrow settings from.
+* @param chunk The DuckDB data chunk to convert.
+* @param out_arrow_array The output Arrow structure that will hold the converted data. Must be released with
+`out_arrow_array->release(out_arrow_array)`
+* @return The error data.
+*/
+DUCKDB_C_API duckdb_error_data duckdb_data_chunk_to_arrow(duckdb_client_properties client_properties,
+                                                          duckdb_data_chunk chunk, struct ArrowArray *out_arrow_array);
+
+/*!
+Transforms an Arrow Schema into a DuckDB Schema.
+
+* @param connection The connection to get the transformation settings from.
+* @param schema The input Arrow schema. Must be released with `schema->release(schema)`.
+* @param out_types The Arrow converted schema with extra information about the Arrow Types. Must be destroyed with
+`duckdb_destroy_arrow_converted_schema`.
+* @param out_names The resulting column names array. Must be appropriately deleted.
```

Comment:
The `out_names` here is problematic for various reasons:

* "Must be appropriately deleted" is unclear 
* It is not actually possible to appropriately delete - given that the out_names are allocated with `new` which has no equivalent in the C API. Anything that requires the user to delete it should be allocated with `duckdb_malloc` so it can be freed with `duckdb_free`
* I wonder if the `out_names` and `out_column_count` are required to begin with? We are already emitting an opaque object (`duckdb_arrow_converted_schema`). Can we not store the names/column count in that opaque object and expose them using accessors?